### PR TITLE
🐛 hack/e2e.sh: retry getting ipaddress object name

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Lot's of e2e CI jobs failed fast with the following error:

```
ipclaim.ipam.metal3.io/ip-claim-0e869fd5082c7413eaadc42d54b26bb27569c203 created
Acquired Control Plane IP: 192.168.6.254
ipclaim.ipam.metal3.io/workload-ip-claim-81ba2a87a03045ce899fe07df55d5f17ae6ce671 created
error: resource name may not be empty
```

`error: resource name may not be empty` comes from trying to get a resource without a name, e.g. `kubectl get pod ""`.

In our `hack/e2e.sh` script this could happen at two places while claiming IP addresses:

When `IPADDRESS_NAME` or `WORKLOAD_IPADDRESS_NAME` are an empty string and using them in one of the following occurencies:

* https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/a9ae01e7a505712399957bce3a20b29f59038df1/hack/e2e.sh#L87

* https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/a9ae01e7a505712399957bce3a20b29f59038df1/hack/e2e.sh#L95

This could happen if the metal3 ipaddress controller did not yet update the status field of the ipclaim, where we read the ipaddress name from.

This PR fixes this by retrying:
* for up to five times
* with a backoff of 1s

to get a non-empty value for the ipaddress name.
It would also output a better error in case of still not being able to read a name from the ipclaim after the timeout.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by chrischdi** [comment]
Ah, shellcheck did already run 🎉

**Comment by sbueringer** [comment]
A bit ironic, golangci-lint
```
 Running [/home/runner/golangci-lint-1.53.3-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
  level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
```

**Comment by chrischdi** [comment]
> A bit ironic, golangci-lint
> 
> ```
>  Running [/home/runner/golangci-lint-1.53.3-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
>   level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
> ```

retriggered by renaming PR 😂

File: hack/e2e.sh [hunk 1]
```diff
@@ -78,20 +78,42 @@ function append_api_group() {
   echo "${resource}.ipam.metal3.io"
 }
 
+function kubectl_get_jsonpath() {
+  local OBJECT_KIND="${1}"
+  local OBJECT_NAME="${2}"
+  local JSON_PATH="${3}"
+  local n=0
+  until [ $n -ge 5 ]; do
```
Comment (sbueringer) [inline hack/e2e.sh:86]: Should we use more retries. Maybe 10 or 30? Just to avoid another itreation because 5s is sometimes not enough

**Comment by sbueringer** [comment]
Thank you very much !!

/lgtm

File: hack/e2e.sh (hunk 1 above)
Comment (chrischdi) [reply hack/e2e.sh:86]: I'll bump to 30 retries.

Normally it should be pretty fast.

**Comment by randomvariable** [comment]
thx for the fix. at some point it'd be good to move to this to [ginkgo.SynchronizedBeforeSuite like there is in the CAPA repo](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/suites/conformance/conformance_suite_test.go#L45C13-L46) instead of bash.

/approve

**Comment by randomvariable** [comment]
Filed https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/2044

**Pull request merged**